snapcraft/internal/lifecycle/_runner.py

```diff
@@ -202,10 +202,17 @@ def run(self, step: steps.Step, part_names=None):
                             getattr(self, '_re{}'.format(
                                 current_step.name))(part)
                         else:
-                            notify_part_progress(
-                                part,
-                                'Skipping {}'.format(current_step.name),
-                                '(already ran)')
+                            outdated_report = self._cache.get_outdated_report(
```

Comment:
I have complications accepting this procedural flow, just the level of nesting.
How is `dirty_report` so different from `outdated_report`? How can we have a `dirty_report` if we haven't `has_step_run`? Do you think this can be shuffled a bit?

Reply:
There, that's quite a bit better. Thoughts?